tsdb/wal/checkpoint.go

```diff
@@ -133,6 +133,13 @@ func Checkpoint(w *WAL, from, to int, keep func(id uint64) bool, mint int64) (*C
 	cpdir := filepath.Join(w.Dir(), fmt.Sprintf(checkpointPrefix+"%06d", to))
 	cpdirtmp := cpdir + ".tmp"
 
+	if _, err := os.Stat(cpdirtmp); !os.IsNotExist(err) {
```

Comment:
This error check feels off. How about always trying, and then use the error check on the removal?

Reply:
ack

Reply:
done